### PR TITLE
Update support menu

### PR DESCRIPTION
**Pull request by hetunandu**

### CODE REVIEW

File: app/client/src/components/designSystems/appsmith/help/HelpModal.tsx [hunk 1]
```diff
@@ -47,48 +44,66 @@ const HelpButton = styled.div<{
   }
 `;
 
+const MODAL_WIDTH = 240;
+const MODAL_HEIGHT = 210;
```
Comment (satbir121) [inline app/client/src/components/designSystems/appsmith/help/HelpModal.tsx:48]: Why was this removed?

File: app/client/src/components/designSystems/appsmith/help/DocumentationSearch.tsx [hunk 2]
```diff
@@ -47,43 +50,75 @@ const StyledDocumentIcon = styled(DocumentIcon)`
...
-  hit: PropTypes.object.isRequired,
 };
 
-const Header = styled.div`
-  position: absolute;
-  width: 100%;
-  border-top-right-radius: 3px;
-  border-top-left-radius: 3px;
-`;
+const HelpMenuItem = (props: {
```
Comment (satbir121) [inline app/client/src/components/designSystems/appsmith/help/DocumentationSearch.tsx:99]: Can we rename this to DefaultHelpMenuItem?

File: app/client/src/components/designSystems/appsmith/help/DocumentationSearch.tsx [hunk 3]
```diff
@@ -205,84 +238,146 @@ const SearchContainer = styled.div`
...
         </InstantSearch>
-      </div>
-    </SearchContainer>
-  );
+      </SearchContainer>
+    );
+  }
 }
+
+const mapStateToProps = (state: AppState) => ({
```
Comment (satbir121) [inline app/client/src/components/designSystems/appsmith/help/DocumentationSearch.tsx:379]: Why did we remove `useSelector`?

File: app/client/src/components/designSystems/appsmith/help/DocumentationSearch.tsx [hunk 4]
```diff
@@ -205,84 +238,146 @@ const SearchContainer = styled.div`
...
+                <ul className="ais-Hits-list">
+                  {HELP_MENU_ITEMS.map(item => (
+                    <HelpMenuItem key={item.label} item={item} />
+                  ))}
+                </ul>
+              )}
+            </HelpBody>
+            <HelpFooter>
+              <span>Appsmith v1.1</span>
+              <span>Released 8 hours ago</span>
```
Comment (satbir121) [inline app/client/src/components/designSystems/appsmith/help/DocumentationSearch.tsx:370]: What's the plan to make `Released 8 hours ago dynamic`?

File: app/client/src/components/designSystems/appsmith/help/HelpModal.tsx (hunk 1 above)
Comment (hetunandu) [reply app/client/src/components/designSystems/appsmith/help/HelpModal.tsx:48]: These values never change. Does not make sense to keep a state for it

File: app/client/src/components/designSystems/appsmith/help/DocumentationSearch.tsx (hunk 4 above)
Comment (hetunandu) [reply app/client/src/components/designSystems/appsmith/help/DocumentationSearch.tsx:370]: These will eventually come from env variables. These are not set as of today so the change will come in a different PR

File: app/client/src/components/designSystems/appsmith/help/DocumentationSearch.tsx [hunk 5]
```diff
@@ -205,84 +238,146 @@ const SearchContainer = styled.div`
...
+
+class DocumentationSearch extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      showResults: props.defaultRefinement.length > 0,
+    };
+  }
+  onSearchValueChange = () => {
+    if (!this.state.showResults) {
```
Comment (satbir121) [inline app/client/src/components/designSystems/appsmith/help/DocumentationSearch.tsx:317]: Shouldn't there be an else block to this which sets `showResults` to false?

File: app/client/src/components/designSystems/appsmith/help/DocumentationSearch.tsx (hunk 3 above)
Comment (hetunandu) [reply app/client/src/components/designSystems/appsmith/help/DocumentationSearch.tsx:412]: We moved to a class component so cant use hooks there

File: app/client/public/index.html [hunk 6]
```diff
@@ -16,6 +16,10 @@
       transition: all ease-in 0.3s;
     }
   </style>
+  <script>
+    // We pre-filled your app ID in the widget URL: 'https://widget.intercom.io/widget/y10e7138'
+    (function(){var w=window;var ic=w.Intercom;if(typeof ic==="function"){ic('reattach_activator');ic('update',w.intercomSettings);}else{var d=document;var i=function(){i.c(arguments);};i.q=[];i.c=function(args){i.q.push(args);};w.Intercom=i;var l=function(){var s=d.createElement('script');s.type='text/javascript';s.async=true;s.src='https://widget.intercom.io/widget/y10e7138';var x=d.getElementsByTagName('script')[0];x.parentNode.insertBefore(s,x);};if(w.attachEvent){w.attachEvent('onload',l);}else{w.addEventListener('load',l,false);}}})();
```
Comment (satbir121) [inline app/client/public/index.html:21]: Do we want to take the api id from environment variables?
